doc/cookbook/open-id-connect/OpenIdConnect.lhs

```diff
@@ -0,0 +1,401 @@
+2.  when the user click on the button it is redirected to the OIDC
+    provider
+3.  the user login in the OIDC provider
+4.  the OIDC provider will redirect the user and provide a `code`
+5.  the server will use this code to make a POST to the OIDC provider
+    and will get back authentication infos
+6.  The user will get display an HTML page that will save a secret
+    identifying him in the local storage, then it will be redirected to
+    /.
+
+First, let's put the import behind us:
+
+``` haskell
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NoImplicitPrelude     #-}
+{-# LANGUAGE OverloadedLists       #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE PartialTypeSignatures #-}
+{-# LANGUAGE PolyKinds             #-}
+{-# LANGUAGE RankNTypes            #-}
+{-# LANGUAGE RecordWildCards       #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE TypeOperators         #-}
+{-# LANGUAGE TypeSynonymInstances  #-}
+
+module Main where
+
+import           Protolude
+
+import           Data.Aeson
+                 (FromJSON (..), (.:))
+import qualified Data.Aeson                       as JSON
+import qualified Data.Aeson.Types                 as AeT
+import qualified Data.ByteString                  as BS
+import qualified Data.ByteString.Lazy             as LBS
+import qualified Data.List                        as List
+import qualified Data.Text                        as Text
+import           Jose.Jwt
+                 (Jwt (..), decodeClaims)
+import           Network.HTTP.Client
+                 (Manager, newManager)
+import           Network.HTTP.Client.TLS
+                 (tlsManagerSettings)
+import           Network.Wai
+                 (Request, requestHeaders)
+import           Network.Wai.Handler.Warp
+                 (run)
+import           Servant
+import           Servant.HTML.Blaze
+                 (HTML)
+import           Servant.Server.Experimental.Auth
+                 (AuthHandler, AuthServerData, mkAuthHandler)
+import qualified System.Random                    as Random
+import           Text.Blaze
+                 (ToMarkup (..))
+import qualified Text.Blaze.Html                  as H
+import           Text.Blaze.Html5
+                 ((!))
+import qualified Text.Blaze.Html5                 as H
+import qualified Text.Blaze.Html5.Attributes      as HA
+import           Text.Blaze.Renderer.Utf8
+                 (renderMarkup)
+import qualified Web.OIDC.Client                  as O
+```
+
+First you'll need to create a new application in Google. Then you should
+fill that OIDCConf for your specific web app.
+
+``` haskell
+oidcConf :: OIDCConf
+oidcConf = OIDCConf { redirectUri = "https://example.org/oauth2/callback"
```

Comment:
It would be nice if you could explain those fields a bit, especially what they map to in your description of the workflow from earlier.